src/services/resource.js

```diff
@@ -206,4 +216,219 @@ module.exports = class resourceHelper {
+					reviewer_id: user_id,
+					organization_id: { [Op.in]: uniqueOrganizationIds },
+					status: { [Op.in]: common.PAGE_STATUS_VALUES.up_for_review },
+				}
+
+				reviewsResponse = await reviewsQueries.findAll(reviewsFilter, commonAttributes)
+
+				if (reviewsResponse.length > 0) {
+					// get the unique organization ids from reviews table by the reviewer
+					uniqueOrganizationIds = [...new Set(reviewsResponse.map((item) => item.organization_id))]
+
+					// get the unique resource ids from reviews table by the reviewer
+					uniqueResourceIds = [...new Set(reviewsResponse.map((item) => item.resource_id))]
+				}
+
+				// fetch all submitted and inreview resources from my orgs
+				const allSequentialResourcesFromOrg = await resourceQueries.findAll(
+					{
+						organization_id,
+						type: { [Op.in]: resourceTypesInSequentialReview },
+						status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+					},
+					['id', 'type', 'next_stage']
+				)
+
+				const openToAllResourcesMatchingMyLevel = allSequentialResourcesFromOrg.map((item) => {
+					if (item.next_stage === resourceWiseLevels[item.type]) return item.id
+				})
+
+				const allParallelResourcesFromOrg = await resourceQueries.findAll(
+					{
+						organization_id,
+						type: { [Op.in]: resourceTypesInParallelReview },
+						status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+					},
+					['id']
+				)
+
+				const allParallelResourceIdsFromOrg = allParallelResourcesFromOrg.map((item) => item.id)
+
+				// remove all the resouces in sequential review picked up by another reviewer
+				reviewsFilter = {
+					reviewer_id: { [Op.notIn]: [user_id] },
+					status: {
+						[Op.in]: [
+							common.REVIEW_STATUS_INPROGRESS,
+							common.REVIEW_STATUS_CHANGES_UPDATED,
+							common.REVIEW_STATUS_REQUESTED_FOR_CHANGES,
+						],
+					},
+					resource_id: { [Op.in]: openToAllResourcesMatchingMyLevel },
+				}
+				reviewsResponse = await reviewsQueries.findAll(reviewsFilter, ['resource_id'])
+
+				// push resource ids to resourceIdsToBeRemoved array
+				resourceIdsToBeRemoved = reviewsResponse.map((item) => item.resource_id)
+
+				finalResourceIds = _.difference(
+					[
+						...new Set([
+							...uniqueResourceIds,
+							...openToAllResourcesMatchingMyLevel,
+							...allParallelResourceIdsFromOrg,
+						]),
+					],
+					resourceIdsToBeRemoved
+				)
+			}
+
+			const resourceFilter = {
+				id: { [Op.in]: finalResourceIds },
+				organization_id,
+			}
+			if (searchText != '')
+				resourceFilter.title = {
+					[Op.iLike]: '%' + searchText + '%',
+				}
+
+			if (common.TYPE in queryParams) {
+				resourceFilter.type = queryParams[common.TYPE]
+			}
+
+			if (
+				common.SORT_BY in queryParams &&
+				common.SORT_ORDER in queryParams &&
+				queryParams.sort_by.length > 0 &&
+				queryParams.sort_order.length > 0
+			) {
+				sort.sort_by = queryParams.sort_by
```

Comment:
@adithyadinesh0412 this can be done in line no 227 itself
const sort = {
            sort_by: queryParams.sort_by || common.CREATED_AT,
            order: queryParams.sort_order?.toUpperCase() === common.SORT_ASC ? common.SORT_ASC : common.SORT_DESC,
        };